flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSinkCompaction.java

```diff
@@ -0,0 +1,180 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.util.DataFormatConverters;
+import org.apache.flink.types.Row;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.ManifestFile;
+import org.apache.iceberg.ManifestFiles;
+import org.apache.iceberg.ManifestReader;
+import org.apache.iceberg.Parameter;
+import org.apache.iceberg.Parameters;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.flink.source.BoundedTableFactory;
+import org.apache.iceberg.flink.source.BoundedTestSource;
+import org.apache.iceberg.flink.util.FlinkCompatibilityUtil;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+
+public class TestFlinkTableSinkCompaction extends CatalogTestBase {
+
+  private static final TypeInformation<Row> ROW_TYPE_INFO =
+      new RowTypeInfo(SimpleDataUtil.FLINK_SCHEMA.getFieldTypes());
+
+  private static final DataFormatConverters.RowConverter CONVERTER =
+      new DataFormatConverters.RowConverter(SimpleDataUtil.FLINK_SCHEMA.getFieldDataTypes());
+
+  private static final String TABLE_NAME = "test_table";
+  private StreamTableEnvironment tEnv;
+  private StreamExecutionEnvironment env;
+  private Table icebergTable;
+  private static final String TABLE_PROPERTIES =
+      "'flink-maintenance.lock.type'='jdbc',"
+          + "'flink-maintenance.lock.jdbc.uri'='jdbc:sqlite:file::memory:?ic',"
+          + "'flink-maintenance.lock.jdbc.init-lock-table'='true',"
+          + "'flink-maintenance.rewrite.rewrite-all'='true',"
+          + "'flink-maintenance.rewrite.schedule.data-file-size'='1',"
+          + "'flink-maintenance.lock-check-delay-seconds'='60'";
+
+  private static final Object[][] TEST_PARAMETERS = {
+    {"testhadoop_basenamespace", Namespace.of("l0", "l1"), true},
+    {"testhadoop_basenamespace", Namespace.of("l0", "l1"), false}
+  };
+
+  @Parameter(index = 2)
+  private boolean userSqlHint = true;
+
+  @Parameters(name = "catalogName={0}, baseNamespace={1}, userSqlHint={2}")
+  public static List<Object[]> parameters() {
+    return Lists.newArrayList(TEST_PARAMETERS);
```

Comment:
We use the parameters only once, so we can put it inline

This is how we usually do this:
```
    return Arrays.asList(
            new Object[] {"testhadoop_basenamespace", Namespace.of("l0", "l1"), true},
            new Object[] {"testhadoop_basenamespace", Namespace.of("l0", "l1"), false});
```